Common/Data/Custom/SmartInsider/SmartInsiderIntention.cs

```diff
@@ -0,0 +1,339 @@
+            ValueCurrency = string.IsNullOrWhiteSpace(csv[47]) ? null : csv[47];
+            IntentionValue = string.IsNullOrWhiteSpace(csv[48]) ? (long?)null : Convert.ToInt64(csv[48], CultureInfo.InvariantCulture);
+            IntentionPercentage = string.IsNullOrWhiteSpace(csv[49]) ? (decimal?)null : Convert.ToDecimal(csv[49], CultureInfo.InvariantCulture);
+            IntentionAuthorisationStartDate = string.IsNullOrWhiteSpace(csv[50]) ? (DateTime?)null : DateTime.ParseExact(csv[50], "yyyy-MM-dd", CultureInfo.InvariantCulture);
+            IntentionAuthorisationEndDate = string.IsNullOrWhiteSpace(csv[51]) ? (DateTime?)null : DateTime.ParseExact(csv[51], "yyyy-MM-dd", CultureInfo.InvariantCulture);
+            PriceCurrency = string.IsNullOrWhiteSpace(csv[52]) ? null : csv[52];
+            MinimumPrice = string.IsNullOrWhiteSpace(csv[53]) ? (decimal?)null : Convert.ToDecimal(csv[53], CultureInfo.InvariantCulture);
+            MaximumPrice = string.IsNullOrWhiteSpace(csv[54]) ? (decimal?)null : Convert.ToDecimal(csv[54], CultureInfo.InvariantCulture);
+            BuybackIntentionNoteText = csv.Length == 56 ? (string.IsNullOrWhiteSpace(csv[55]) ? null : csv[55]) : null;
+        }
+
+        /// <summary>
+        /// Specifies the location of the data and directs LEAN where to load the data from
+        /// </summary>
+        /// <param name="config">Subscription configuration</param>
+        /// <param name="date">Algorithm date</param>
+        /// <param name="isLiveMode">Is live mode</param>
+        /// <returns>Subscription data source object pointing LEAN to the data location</returns>
+        public override SubscriptionDataSource GetSource(SubscriptionDataConfig config, DateTime date, bool isLiveMode)
+        {
+            return new SubscriptionDataSource(
+                Path.Combine(
+                    Globals.DataFolder,
+                    "alternative",
+                    "smartinsider",
+                    "intentions",
+                    $"{config.Symbol.Value.ToLower()}.csv"
+                ),
+                SubscriptionTransportMedium.LocalFile,
+                FileFormat.Csv
+            );
+        }
+
+        /// <summary>
+        /// Loads and reads the data to be used in LEAN
+        /// </summary>
+        /// <param name="config">Subscription configuration</param>
+        /// <param name="line">CSV line</param>
+        /// <param name="date">Algorithm date</param>
+        /// <param name="isLiveMode">Is live mode</param>
+        /// <returns>Instance of the object</returns>
+        public override BaseData Reader(SubscriptionDataConfig config, string line, DateTime date, bool isLiveMode)
+        {
+            var intention = new SmartInsiderIntention(line)
+            {
+                Symbol = config.Symbol
+            };
+            intention.Time = intention.Time.AddHours(17).ConvertTo(TimeZones.London,config.DataTimeZone);
```

Comment:
Maybe a comment describing why 17 hours?

Reply:
I added a comment in the CSV line constructor, but I agree that it's a good idea to have it here as well